core/server/analytics.js

```diff
@@ -117,27 +117,67 @@ Meteor.methods({
 
     analytics.updateUser(editedUserId, traits);
   },
-  analyticsDiscussionAttend(traits) {
+  analyticsDiscussionStart(properties) {
     const { userId } = this;
     if (!userId) return;
 
-    check(traits, {
-      users_attending_count: Number,
+    check(properties, {
+      peerUserId: String,
     });
 
     const user = Meteor.user();
-    analytics.track(userId, '💬 Discussion Attend', { level_id: user.profile.levelId });
+    analytics.track(userId, '💬 ✅ Discussion Start', {
+      level_id: user.profile.levelId,
+      peer_user_id: properties.peerUserId,
+    });
   },
-  analyticsDiscussionEnd(traits) {
+  analyticsDiscussionJoin(properties) {
     const { userId } = this;
     if (!userId) return;
 
-    check(traits, {
+    check(properties, {
+      peerUserId: String,
+    });
+
+    const user = Meteor.user();
+    analytics.track(userId, '💬 ➡️ Discussion Join', {
+      level_id: user.profile.levelId,
+      peer_user_id: properties.peerUserId,
+    });
+  },
+  analyticsDiscussionEnd(properties) {
+    const { userId } = this;
+    if (!userId) return;
+
+    check(properties, {
+      totalDuration: Number,
       duration: Number,
```

Comment:
I don't understand the difference between `duration` and `totalDuration`; there's one for the call and the other for the discussion? If so, I don't understand why we need to send both for the Discussion End.

Reply:
Sorry, the answer is in the commit message. Can't we just put `totalDuration` as `duration` and remove `duration` for this event in particular?

Reply:
No we cannot because the total duration of the discussion is not necessarily the duration of the call with the last user leaving (or in other words the last user leaving is not necessarily the one who started the discussion).
The other possibility is to throw a Discussion Left event alongside the Discussion End but I thought it was a bit heavy to send 2 events for each discussion end (which are most of the time only between two peoples).